internal/cli/kraft/lib/add/add.go

```diff
@@ -0,0 +1,171 @@
+}
+
+// Add adds a Unikraft library to the project directory and updates the Kraftfile
+func Add(ctx context.Context, opts *AddOptions, args ...string) error {
+	if opts == nil {
+		opts = &AddOptions{}
+	}
+
+	return opts.Run(ctx, args)
+}
+
+func NewCmd() *cobra.Command {
+	cmd, err := cmdfactory.New(&AddOptions{}, cobra.Command{
+		Short: "Add unikraft library to the project directory",
+		Use:   "add [FLAGS] [PACKAGE|DIR]",
+		Args:  cmdfactory.MinimumArgs(1, "library name is not specified"),
+		Long: heredoc.Doc(`
+			Pull a Unikraft component microlibrary from a remote location
+			and add to the project directory
+		`),
+		Annotations: map[string]string{
+			cmdfactory.AnnotationHelpGroup: "lib",
+		},
+		Example: heredoc.Doc(`
+			# Add a local library to the project
+			$ kraft lib add path/to/library
+
+			# Add a library from a source repository
+			$ kraft lib add github.com/unikraft/lib-nginx.git
+
+			# Add from a manifest
+			$ kraft lib add nginx:staging
+
+			# Add a library from a registry
+			$ kraft lib add unikraft.org/nginx:stable
+		`),
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	return cmd
+}
+
+func (opts *AddOptions) Pre(cmd *cobra.Command, _ []string) error {
+	ctx, err := packmanager.WithDefaultUmbrellaManagerInContext(cmd.Context())
+	if err != nil {
+		return err
+	}
+	cmd.SetContext(ctx)
+	return nil
+}
+
+func (opts *AddOptions) Run(ctx context.Context, args []string) error {
+	var workdir, packName, packVersion string
+	var packType unikraft.ComponentType
+	var err error
+	var library lib.LibraryConfig
+	packageManager := packmanager.G(ctx)
+
+	if f, err := os.Stat(args[0]); err == nil && f.IsDir() {
+		if err = packageManager.AddSource(ctx, args[0]); err != nil {
+			return err
+		}
+		config.G[config.KraftKit](ctx).Unikraft.Manifests = append(
+			config.G[config.KraftKit](ctx).Unikraft.Manifests,
+			args[0],
+		)
+		if err := config.M[config.KraftKit](ctx).Write(true); err != nil {
+			return err
+		}
+		if err = packageManager.Update(ctx); err != nil {
+			return err
+		}
+		tempStrs := strings.Split(args[0], "/")
+		packName = tempStrs[len(tempStrs)-1]
+		packVersion = "default"
+		packType = unikraft.ComponentTypeUnknown
+		args[0] = packName + ":" + packVersion
+	} else {
+		// Verifying if user demands for a library otherwise return error.
+		packType, packName, packVersion, err = unikraft.GuessTypeNameVersion(args[0])
+		if err != nil {
+			tempStrs := strings.Split(args[0], "/")
+			tempStr := tempStrs[len(tempStrs)-1]
+
+			packType, packName, packVersion, err = unikraft.GuessTypeNameVersion(tempStr)
+			if err != nil {
+				return err
+			}
+		}
+
+		if len(packType) > 0 && packType != unikraft.ComponentTypeLib && packType != unikraft.ComponentTypeUnknown {
+			return fmt.Errorf("specified package %s is not a library", args[0])
+		}
+	}
+
+	if len(packType) > 0 && !strings.HasPrefix(args[0], "lib") {
+		args[0] = "lib/" + args[0]
+	}
```

Comment:
This fails when trying to pull packages because it prepends `lib/` on urls :(

Also this did not work for me:
```
kraft lib add github.com/unikraft/lib-nginx.git

 D  kraftkit v0.7.1-36-g78e0abd-6-g2ad720ea
 T  initializing package manager format=oci
 T  using oci directory handler path=/home/cez/.local/share/kraftkit/runtime/oci
 T  initializing package manager format=manifest
 E  cannot determine name and type from "lib-nginx.git"
```